astropy/time/tests/test_methods.py

```diff
@@ -620,6 +629,37 @@ def test_sort(self, use_mask):
             assert np.all(self.t0.sort(-1)[:, :, 0] == self.t0.min(-1))
             assert np.all(self.t0.sort(-1)[:, :, -1] == self.t0.max(-1))
 
+    @pytest.mark.parametrize('axis', [None, 0, 1, 2, (0, 1)])
+    @pytest.mark.parametrize('where', [True, np.array([True, False, True, True, False])])
+    def test_mean(self, use_mask, axis, where):
+        self.create_data(use_mask)
+
+        kw = dict(axis=axis, where=where, keepdims=True)
+
+        def is_consistent(time):
+            time_mean = time.mean(**kw)
+            value_test = time_mean.jd1 + time_mean.jd2
+            value_expected = Time(time.tai.jd1.mean(**kw), time.tai.jd2.mean(**kw), format='jd', scale='tai')
+            value_expected._set_scale(time.scale)
+            value_expected = value_expected.jd1 + value_expected.jd2
+            return np.all(value_test == value_expected)
+
+        if use_mask == 'not_masked':
+            assert is_consistent(self.t0)
+            assert is_consistent(self.t1)
+
+            if axis in [None, 2]:
+                with pytest.raises(ValueError):
+                    self.t2.mean(**kw)
+            else:
+                assert is_consistent(self.t2)
+
+        else:
+            with pytest.raises(ValueError):
+                self.t0.mean()
+            with pytest.raises(ValueError):
+                self.t1.mean()
+
```

Comment:
Could you also add a test that checks that that a `where` that causes empty slices leads to a `ValueError`. And, just for completeness, that `dtype` and `out` cannot be passed in?